alerter.c

```diff
@@ -1,31 +1,65 @@
-#include <stdio.h>
-#include <assert.h>
+#include "alerter.h"
 
 int alertFailureCount = 0;
 
 int networkAlertStub(float celcius) {
-    printf("ALERT: Temperature is %.1f celcius.\n", celcius);
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
+    
+        printf("Temperature is %.1f celcius.\n", celcius);
+        return 200;
+   
 }
 
-void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
+int networkAlertProduction(float celcius) {
+    
+    if(celcius < 50.0)
+    {
+        printf("Temperature is %.1f celcius.\n", celcius);
+        return 200;
+    }
+    else
+    {
+        printf("ALERT: Temperature is %.1f celcius.\n", celcius);
+        return 500;
+    }
+}
+
+float tempConversion(float farenheit)
+{
+  float celcius;
+  celcius = (farenheit - 32) * 5 / 9;
+  return celcius; 
+}
+
+void alertInCelcius(float farenheit, testEnvironment environment) {
+    float celcius = tempConversion(farenheit);
+    int returnCode;
+    if(environment == networkStub)

```

Comment:
Do not mix production code with test code. Keep them separate. The production code should not be aware of test environment and hence  handle differently in test environment. The definite set of input should result in definite set of output. That's the functionality of production code. What is the intention to keep test stub and production code (networkAlertProduction) in the same file?